tf2onnx/function/gathernd.py

```diff
@@ -0,0 +1,127 @@
+    body_nodes = []
+    index_i = ctx.make_node("Gather", [index.output[0], "inner_i"], attr={"axis": 0})
+    gather = ctx.make_node("Gather", ["gathernd_cur", index_i.output[0]], attr={"axis": 0})
+    squeeze = ctx.make_node("Squeeze", [gather.output[0]], attr={"axes": [0]}, outputs=["inner_res"])
+    body_nodes.extend([index_i.op, gather.op, squeeze.op])
+    body_nodes.append(utils.make_onnx_identity("inner_cond", "inner_cond_out"))
+    body_graph = helper.make_graph(body_nodes, utils.make_name("gathernd_inner_body"), body_inputs, body_outputs)
+    inner_loop = ctx.make_node("Loop", [trip_node.output[0],
+                                        cond_const.output[0],
+                                        params],
+                               attr={"body": body_graph})
+    nodes.append(inner_loop.op)
+    return nodes, inner_loop
+
+def gathernd_op(ctx, node, name, args):
+    """GatherNd op."""
+    # T output = GatherNd(T Input, INT32/INT64 indices)
+    nodes = []
+    params = node.input[0]
+    indices = node.input[1]
+    # same as the attr Tparams
+    node_dtype = ctx.get_dtype(params)
+    utils.make_sure(node_dtype, "Dtype of {} is None".format(indices))
+    # reshape indices into [sum(indices[:-1]), indices[-1]]
+    indices_shape = ctx.make_node("Shape", [indices], dtypes=[TensorProto.INT64])
+    outter_shape = ctx.make_node("Slice",
+                                 [indices_shape.output[0]],
+                                 attr={"axes": [0], "ends": [-1], "starts": [0]},
+                                 dtypes=[TensorProto.INT64])
+    inner_shape = ctx.make_node("Slice",
+                                [indices_shape.output[0]],
+                                attr={"axes": [0], "ends": [sys.maxsize], "starts": [-1]},
+                                dtypes=[TensorProto.INT64])
+    outter_shape_sum = ctx.make_node("ReduceSum",
+                                     [outter_shape.output[0]],
+                                     attr={"axes": [0], "keepdims": 1},
+                                     dtypes=[TensorProto.INT64])
+    flatten_shape = ctx.make_node("Concat",
+                                  [outter_shape_sum.output[0], inner_shape.output[0]],
+                                  attr={"axis": 0},
+                                  dtypes=[TensorProto.INT64])
+    flatten_indices = ctx.make_node("Reshape", [indices, flatten_shape.output[0]])
+    nodes.extend([indices_shape, outter_shape, inner_shape, outter_shape_sum, flatten_shape, flatten_indices])
+    # outter loop for each index
+    # for (int i=0; i<outter_shape_sum; i++) inner_loop(params, flatten_indices[i])
+    cond_const = ctx.make_const(utils.make_name("cond"), np.ones((), dtype=np.bool))
+    dummy_const = ctx.make_const(utils.make_name("dummy"), np.ones((), dtype=np.int64))
+    body_inputs = [helper.make_tensor_value_info("i", onnx_pb.TensorProto.INT64, []),
```

Comment:
better not hardcode the input name, use make_name for "i", in case of naming conflict.

Reply:
same for output